code/game/objects/items/weapons/RCD.dm

```diff
@@ -144,239 +218,229 @@ RCD
-				if(useResource(1, user))
-					to_chat(user, "Building Floor...")
-					activate()
-					var/turf/AT = get_turf(A)
-					AT.ChangeTurf(/turf/simulated/floor/plating)
-					return 1
-				return 0
-
-			if(istype(A, /turf/simulated/floor))
-				if(checkResource(3, user))
-					to_chat(user, "Building Wall ...")
-					playsound(loc, 'sound/machines/click.ogg', 50, 1)
-					if(do_after(user, 20 * toolspeed, target = A))
-						if(!useResource(3, user))
-							return 0
-						activate()
-						var/turf/AT = A
-						AT.ChangeTurf(/turf/simulated/wall)
-						return 1
-				return 0
-
-		if(2)
-			if(istype(A, /turf/simulated/floor))
-				if(checkResource(10, user))
-					to_chat(user, "Building Airlock...")
-					playsound(loc, 'sound/machines/click.ogg', 50, 1)
-					if(do_after(user, 50 * toolspeed, target = A))
-						for(var/obj/machinery/door/airlock/D in A.contents)
-							return 0
-						if(!useResource(10, user))
-							return 0
-						activate()
-						var/obj/machinery/door/airlock/T = new door_type(A)
-						T.name = door_name
-						T.autoclose = 1
-						if(one_access)
-							T.req_one_access = door_accesses.Copy()
-						else
-							T.req_access = door_accesses.Copy()
-						return 1
-					return 0
-				return 0
-
-		if(3)
-			if(istype(A, /turf/simulated/wall))
-				if(istype(A, /turf/simulated/wall/r_wall) && !canRwall)
-					return 0
-				if(checkResource(5, user))
-					to_chat(user, "Deconstructing Wall...")
-					playsound(loc, 'sound/machines/click.ogg', 50, 1)
-					if(do_after(user, 40 * toolspeed, target = A))
-						if(!useResource(5, user))
-							return 0
-						activate()
-						var/turf/AT = A
-						AT.ChangeTurf(/turf/simulated/floor/plating)
-						return 1
-				return 0
-
-			if(istype(A, /turf/simulated/floor))
-				if(checkResource(5, user))
-					to_chat(user, "Deconstructing Floor...")
-					playsound(loc, 'sound/machines/click.ogg', 50, 1)
-					if(do_after(user, 50 * toolspeed, target = A))
-						if(!useResource(5, user))
-							return 0
-						activate()
-						var/turf/AT = A
-						AT.ChangeTurf(/turf/space)
-						return 1
-				return 0
-
-			if(istype(A, /obj/machinery/door/airlock))
-				if(checkResource(20, user))
-					to_chat(user, "Deconstructing Airlock...")
-					playsound(loc, 'sound/machines/click.ogg', 50, 1)
-					if(do_after(user, 50 * toolspeed, target = A))
-						if(!useResource(20, user))
-							return 0
-						activate()
-						qdel(A)
-						return 1
-				return	0
-
-			if(istype(A, /obj/structure/window)) // You mean the grille of course, do you?
-				A = locate(/obj/structure/grille) in A.loc
-
-			if(istype(A, /obj/structure/grille))
-				if(!checkResource(2, user))
-					return 0
-				to_chat(user, "Deconstructing window...")
-				playsound(loc, 'sound/machines/click.ogg', 50, 1)
-				if(!do_after(user, 20 * toolspeed, target = A))
-					return 0
-				if(locate(/obj/structure/window/full/shuttle) in A.contents)

```

Comment:
I also did not readd this, because it too, doesn't work. It's checking for a window... *inside a grill*, not on it's turf.